### PR TITLE
Add ScalarField class and sphere generation code

### PR DESCRIPTION
**Pull request by abhimadan**
The scalar field is surfaced using the marching cubes algorithm. The
script to generate the table it uses isn't very readable, unfortunately.

### CODE REVIEW

**Comment by andrew749** [comment]
Those are a lot of lint error. you should probably fix some of these up.

**Comment by andrew749** [comment]
`yarn fix-format` should do it.

**Comment by abhimadan** [comment]
I ran `yarn fix-format` before sending this - these errors aren't ones that it can fix. They're fixed now though.

File: src/geometry/Sphere.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+const RADIUS = 1;
+const DIM = 25;
+const LENGTH = 2.5;
+
+function sphereSignedDistFunc(coord: vec3): number {
+    return Math.sqrt(coord[0] * coord[0] + coord[1] * coord[1] + coord[2] * coord[2]) - RADIUS;
+}
+
+// TODO(abhimadan): Create WorkingGeometry here instead
```
Comment (davepagurek) [inline src/geometry/Sphere.ts:16]: Can we make an issue for this?

File: src/geometry/MarchingCubes.ts [hunk 2]
```diff
@@ -0,0 +1,318 @@
+import { vec3 } from 'gl-matrix';
+import { range } from 'lodash';
+
+import { ScalarField } from './ScalarField';
+
+const triTable: number[][] = [
```
Comment (davepagurek) [inline src/geometry/MarchingCubes.ts]: Some comments just explaining what this gets used for would be great (maybe a URL to an article explaining it?)

File: src/geometry/ScalarField.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                        });
+                    });
+
+                    func(voxel, xIdx, yIdx, zIdx);
+                });
+            });
+        });
+    }
+
+    public indexToModel(x: number, y: number, z: number): vec3 {
```
Comment (davepagurek) [inline src/geometry/ScalarField.ts]: What is this an index of?

**Comment by davepagurek** [comment]
Good stuff! Just could use some comments for things we're expecting to interface with (the public methods of ScalarField I guess)

File: src/geometry/Sphere.ts (hunk 1 above)
Comment (abhimadan) [reply src/geometry/Sphere.ts:16]: See #43.

File: src/geometry/MarchingCubes.ts (hunk 2 above)
Comment (abhimadan) [reply src/geometry/MarchingCubes.ts]: Done.

File: src/geometry/ScalarField.ts (hunk 3 above)
Comment (abhimadan) [reply src/geometry/ScalarField.ts]: It converts an index into the scalar field array to a coordinate in the scalar field's model space. I added a comment to explain this.

**Review by davepagurek — APPROVED** [review]

**Pull request merged**